### PR TITLE
Run System.Net.Http trimming tests with AOT

### PR DESCRIPTION
**Pull request by MichalStrehovsky**
One is not compatible due to a known bug.

### CODE REVIEW

File: src/libraries/System.Net.Http/tests/TrimmingTests/System.Net.Http.TrimmingTests.proj [hunk 1]
```diff
@@ -5,6 +5,8 @@
     <TestConsoleAppSourceFiles Include="DecompressionHandlerTrimmedTest.cs" />
     <TestConsoleAppSourceFiles Include="HttpClientTest.cs">
       <SkipOnTestRuntimes>browser-wasm</SkipOnTestRuntimes>
+      <!-- https://github.com/dotnet/runtime/issues/73290 -->
```
Comment (eerhardt) [inline src/libraries/System.Net.Http/tests/TrimmingTests/System.Net.Http.TrimmingTests.proj:8]: Looking at the test:

https://github.com/dotnet/runtime/blob/40bc2d88b6606324f5774cc972d480a1d26084f8/src/libraries/System.Net.Http/tests/TrimmingTests/HttpClientTest.cs#L24-L25

I'm not sure how the test would ever work on NativeAOT even if this bug is fixed. It is expecting a managed .dll to exist on disk.

Comment (MichalStrehovsky) [reply src/libraries/System.Net.Http/tests/TrimmingTests/System.Net.Http.TrimmingTests.proj:8]: Huh, I thought it's checking for the native DLL. I guess I don't understand what this is testing for.

```suggestion
      <!-- Checks for presence of managed assemblies on disk -->
```

Comment (eerhardt) [reply src/libraries/System.Net.Http/tests/TrimmingTests/System.Net.Http.TrimmingTests.proj]: > I guess I don't understand what this is testing for.

https://github.com/dotnet/runtime/pull/49261/files#r611752507

**Review by eerhardt — APPROVED** [review]
LGTM

**Pull request merged**